### DISCUSSION TITLE
Several users upload files with identical name (different directories), how to add the username to the file name and copy it?

### DISCUSSION THREAD

**Discussion by Batnun** [drakkan/sftpgo#1277] [Q&A]
Hi,

I have a server where several users (different usernames) upload files with an identical filename (to different directories).
How can I rename the uploaded file automatically by adding the username to the filename, and then copy/move the file to another directory/share?

Thanks!

**Comment by drakkan** [comment]
Hello,

I think this can be done using the EventManager. I'm sorry but I no longer provide free step by step support. If you need support, consider [sponsoring](https://github.com/sponsors/drakkan) the project or purchasing a support plan from the [SFTPGo website](https://sftpgo.com/#pricing). Thank you

**Comment by GiladHi** [comment]
Also interested in the solution to this question...

**Comment by drakkan** [comment]
> Also interested in the solution to this question...

this should be possible using the EventManager, take a look at the recycle example [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/eventmanager.md#recycle-bin). It should be quite similar, just use the copy action instead of the rename action
